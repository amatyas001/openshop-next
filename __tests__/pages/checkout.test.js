import { create, act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Checkout from '@app/pages/checkout';

const mockStore = configureStore([]);

describe('<Checkout />', () => {
  let tree, store;

  const status = ['review', 'form', 'confirm', 'success', 'cancelled', 'error'];

  it('should render unauthorized component', async () => {
    await act(async () => {
      tree = create(
        <Provider store={mockStore({})}>
          <Checkout />
        </Provider>
      );
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });

  status.map((item) => {
    it(`should render ${item} component`, async () => {
      store = mockStore({
        cart: [
          {
            id: 'mock_id',
            name: 'mock_name',
            description: 'mock_desc',
            img: 'mock_img',
            price: 10,
            starrating: 5,
            color: 'mock_color',
            amount: 10,
            buy: {
              amount: 1,
            },
          },
        ],
        payment: {
          status: item,
          intent: { id: 'id' },
          error: { message: 'message' },
          details: { name: 'name' },
        },
      });

      await act(async () => {
        tree = create(
          <Provider store={store}>
            <Checkout />
          </Provider>
        );
      });

      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
