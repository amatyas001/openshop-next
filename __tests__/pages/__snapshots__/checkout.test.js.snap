// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkout /> should render cancelled component 1`] = `
<div
  className="css-1tfhyzp"
>
  <div
    className="css-1sj67m5"
  >
    <p
      className="css-18poma1"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
    >
      Review
    </p>
    <svg
      className="css-0"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
      />
    </svg>
    <p
      className="css-18poma1"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
    >
      Payment Details
    </p>
    <svg
      className="css-0"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
      />
    </svg>
    <p
      className="css-18poma1"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
    >
      Finished
    </p>
  </div>
  <div
    className="css-1mi3tt8"
  >
    <h2
      className="css-84al3w"
    >
      Payment Cancelled
    </h2>
    <p
      className="css-441ftm"
    >
      Successfuly cancelled intent: 
      <span
        className="css-4i139l"
        data-testid="cancelled-details"
      >
        id
      </span>
    </p>
  </div>
</div>
`;

exports[`<Checkout /> should render confirm component 1`] = `
<div
  className="css-1tfhyzp"
>
  <div
    className="css-1sj67m5"
  >
    <p
      className="css-18poma1"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
    >
      Review
    </p>
    <svg
      className="css-0"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
      />
    </svg>
    <p
      className="css-18poma1"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
    >
      Payment Details
    </p>
    <svg
      className="css-0"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
      />
    </svg>
    <p
      className="css-18poma1"
      style={Object {}}
    >
      Finished
    </p>
  </div>
  <div
    className="css-1mi3tt8"
  >
    <h2
      className="css-ijdyq2"
      data-testid="confirm-intent-name"
    >
      Hello, name!
    </h2>
    <p
      className="css-tmw1y6"
    >
      Please review your payment details and confirm order
    </p>
    <div
      className="css-r96hqv"
    >
      <p
        className="css-18poma1"
      >
        INTENT ID
      </p>
      <p
        className="css-1n4bukm"
        data-testid="confirm-intent-id"
      >
        id
      </p>
      <p
        className="css-18poma1"
      >
        EMAIL
      </p>
      <p
        className="css-1n4bukm"
        data-testid="confirm-intent-email"
      />
      <p
        className="css-18poma1"
      >
        PHONE
      </p>
      <p
        className="css-1n4bukm"
        data-testid="confirm-intent-phone"
      />
      <p
        className="css-18poma1"
      >
        ADDRESS
      </p>
      <p
        className="css-1n4bukm"
        data-testid="confirm-intent-address"
      />
    </div>
    <ul
      className="css-8atqhb"
      role="list"
    >
      <li
        className="css-1tp0oa"
        role="listitem"
      >
        <img
          alt="mock_name"
          className="css-rhemcv"
        />
        <div
          className="css-zbq0po"
        >
          <strong
            className="css-airqvi"
          >
            mock_name
          </strong>
          <em
            className="css-1fji79j"
          >
            mock_color in One Size x1
          </em>
        </div>
        <strong
          className="css-1som8bm"
        >
          10.00
           $
        </strong>
      </li>
    </ul>
    <Elements
      stripe="stripe"
    >
      <div
        className="css-1mi3tt8"
      >
        <label
          className="css-8qu6k5"
          htmlFor="StripeElement"
        >
          enter your card details to finish payment
        </label>
        <div
          className="css-16soxp"
        >
          <CardElement
            data-testid="confirm-card-element"
            onChange={[Function]}
            options={
              Object {
                "iconStyle": "solid",
                "style": Object {
                  "base": Object {
                    "fontFamily": "Khand, sans-serif",
                    "fontSize": "18px",
                    "fontSmoothing": "antialiased",
                    "fontWeight": "bold",
                  },
                },
              }
            }
          />
        </div>
        <p
          className="css-165naot"
        >
          Use any future date and CVC with
          <strong>
             4242 4242 4242 4242 
          </strong>
          number for testing
        </p>
      </div>
    </Elements>
    <div
      className="css-qv6zkf"
    >
      <button
        className="css-1e5j5cf"
        data-testid="confirm-controls-cancel"
        href="#void"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        CANCEL ORDER
      </button>
      <button
        className="css-hn65qy"
        data-testid="confirm-controls-confirm"
        disabled={true}
        href="#void"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        CONFIRM ORDER
      </button>
    </div>
  </div>
  <div
    className="css-qqpb0e"
    data-testid="confirm-spinner"
  >
    <h2
      className="css-1gjlikv"
    >
      Loading data...
    </h2>
    <div
      className="css-159m1lu"
    >
      <div
        className="css-1gy22ay"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;

exports[`<Checkout /> should render error component 1`] = `
<div
  className="css-1tfhyzp"
>
  <div
    className="css-1sj67m5"
  >
    <p
      className="css-18poma1"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
    >
      Review
    </p>
    <svg
      className="css-0"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
      />
    </svg>
    <p
      className="css-18poma1"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
    >
      Payment Details
    </p>
    <svg
      className="css-0"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
      />
    </svg>
    <p
      className="css-18poma1"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
    >
      Finished
    </p>
  </div>
  <div
    className="css-1mi3tt8"
  >
    <p
      className="css-1po2p9s"
    >
      <strong>
        Error:
      </strong>
      <br />
      <em
        data-testid="error-message"
      >
        message
      </em>
    </p>
    <button
      className="css-hn65qy"
      data-testid="error-button-retry"
      href="#void"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex="0"
      type="button"
    >
      retry payment process
    </button>
  </div>
</div>
`;

exports[`<Checkout /> should render form component 1`] = `
<div
  className="css-1tfhyzp"
>
  <div
    className="css-1sj67m5"
  >
    <p
      className="css-18poma1"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
    >
      Review
    </p>
    <svg
      className="css-0"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
      />
    </svg>
    <p
      className="css-18poma1"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
    >
      Payment Details
    </p>
    <svg
      className="css-0"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
      />
    </svg>
    <p
      className="css-18poma1"
      style={Object {}}
    >
      Finished
    </p>
  </div>
  <form
    className="css-8o331q"
  >
    <fieldset
      className="css-94sf3x"
    >
      <fieldset
        className="css-9x3bhi"
      >
        <label
          className="css-1cj26ek"
          htmlFor="name"
        >
          name
        </label>
        <input
          autoComplete={false}
          className="css-185v6vq"
          id="name"
          onChange={[Function]}
          placeholder="John Doe"
          required={true}
          type="text"
          value=""
        />
      </fieldset>
      <fieldset
        className="css-9x3bhi"
      >
        <label
          className="css-1cj26ek"
          htmlFor="address"
        >
          address
        </label>
        <input
          autoComplete={false}
          className="css-185v6vq"
          id="address"
          onChange={[Function]}
          placeholder="J191 S. Illinois Street, Everett, MA 02149"
          required={true}
          type="text"
          value=""
        />
      </fieldset>
      <fieldset
        className="css-9x3bhi"
      >
        <label
          className="css-1cj26ek"
          htmlFor="email"
        >
          email
        </label>
        <input
          autoComplete={false}
          className="css-185v6vq"
          id="email"
          onChange={[Function]}
          placeholder="johndoe@example.com"
          required={true}
          type="email"
          value=""
        />
      </fieldset>
      <fieldset
        className="css-9x3bhi"
      >
        <label
          className="css-1cj26ek"
          htmlFor="phone"
        >
          phone
        </label>
        <input
          autoComplete={false}
          className="css-185v6vq"
          id="phone"
          onChange={[Function]}
          placeholder="(941) 555-0123"
          required={true}
          type="tel"
          value=""
        />
      </fieldset>
    </fieldset>
    <p
      className="css-1yab5hb"
    >
      By proceeding you accept the
      <a
        className="css-1c5zfe9"
        href="#void"
      >
         Terms of Service 
      </a>
      of the company.
    </p>
    <div
      className="css-1rsrz83"
    >
      <button
        className="css-1h0dud8"
        data-testid="form-review-button"
        href="#void"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        BACK TO REVIEW
      </button>
      <button
        className="css-hn65qy"
        data-testid="form-submit-button"
        disabled={true}
        href="#void"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        FILL THE FORM
      </button>
    </div>
  </form>
</div>
`;

exports[`<Checkout /> should render review component 1`] = `
<div
  className="css-1tfhyzp"
>
  <div
    className="css-1sj67m5"
  >
    <p
      className="css-18poma1"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
    >
      Review
    </p>
    <svg
      className="css-0"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
      />
    </svg>
    <p
      className="css-18poma1"
      style={Object {}}
    >
      Payment Details
    </p>
    <svg
      className="css-0"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
      />
    </svg>
    <p
      className="css-18poma1"
      style={Object {}}
    >
      Finished
    </p>
  </div>
  <div
    className="css-1djihfe"
  >
    <ul
      className="css-7822g1"
      role="list"
    >
      <li
        className="css-1tp0oa"
        role="listitem"
      >
        <img
          alt="mock_name"
          className="css-rhemcv"
        />
        <div
          className="css-zbq0po"
        >
          <strong
            className="css-airqvi"
          >
            mock_name
          </strong>
          <em
            className="css-1fji79j"
          >
            mock_color in One Size x1
          </em>
          <p
            className="css-18poma1"
            data-testid="cart-item-description"
          >
            mock_desc
          </p>
        </div>
        <strong
          className="css-1som8bm"
        >
          10.00
           $
        </strong>
      </li>
    </ul>
    <div
      className="css-1jgiy2l"
    >
      <button
        className="css-1wjk1cz"
        data-testid="review-button-back"
        href="#void"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        BACK
      </button>
      <button
        className="css-1ow8n45"
        data-testid="review-button-form"
        href="#void"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
        type="button"
      >
        PAYMENT DETAILS
      </button>
      <div
        className="css-ufscxh"
      >
        <p
          className="css-dptndp"
        >
          TOTAL
        </p>
        <strong
          className="css-lk14dv"
          data-testid="cart-amount-value"
        >
          10.00
           $
        </strong>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Checkout /> should render success component 1`] = `
<div
  className="css-1tfhyzp"
>
  <div
    className="css-1sj67m5"
  >
    <p
      className="css-18poma1"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
    >
      Review
    </p>
    <svg
      className="css-0"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
      />
    </svg>
    <p
      className="css-18poma1"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
    >
      Payment Details
    </p>
    <svg
      className="css-0"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z"
      />
    </svg>
    <p
      className="css-18poma1"
      style={
        Object {
          "color": "#44337A",
          "fontWeight": "bold",
        }
      }
    >
      Finished
    </p>
  </div>
  <div
    className="css-1mi3tt8"
  >
    <h2
      className="css-84al3w"
    >
      Payment Successful
    </h2>
    <p
      className="css-441ftm"
    >
      Received intent number:
      <span
        className="css-4i139l"
        data-testid="success-content"
      >
        &nbsp;id&nbsp;
      </span>
    </p>
    <p
      className="css-18poma1"
    >
      This was a sample transaction to demonstrate the payment flow. Thank you!
    </p>
  </div>
</div>
`;

exports[`<Checkout /> should render unauthorized component 1`] = `
<div
  className="css-1tfhyzp"
>
  
  <div
    className="css-dvxtzn"
  >
    <h2
      className="css-84al3w"
    >
      You are not authorized to view this page!
    </h2>
    <p
      className="css-18poma1"
    >
      Please start the payment process from your cart!
    </p>
  </div>
</div>
`;
